src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -1,100 +1,121 @@
+import {
+  containerStyle,
+  dotStyle,
+  messageStyle,
+  styles,
+  textContainerStyle,
+  textStyle
+} from './ChatItemText.style';
 
-const {width} = Dimensions.get('screen');
-const styles = StyleSheet.create({
-  chatContainer: {
-    display: 'flex',
-    flexDirection: 'row',
-    marginTop: 4,
-    marginBottom: 4,
-    maxWidth: width,
-    paddingRight: 60,
-    paddingLeft: 10
-  },
-  chatTitleContainer: {
-    display: 'flex',
-    flexDirection: 'row'
-  },
-  textContainer: {
-    backgroundColor: colors.lightgrey,
-    paddingLeft: 8,
-    paddingRight: 8,
-    paddingTop: 4,
-    paddingBottom: 8,
-    borderRadius: 8,
-    borderTopStartRadius: 0,
-    flex: 1
-  },
-  userText: {
-    fontFamily: fonts.inter[600],
-    fontSize: 12,
-    lineHeight: 19.36
-  },
-  text: {
-    fontFamily: fonts.inter[400],
-    fontSize: 16,
-    lineHeight: 19.36
-  },
-  avatar: {
-    width: 24,
-    height: 24,
-    borderRadius: 12,
-    marginRight: 8
-  },
-  dot: {
-    width: 3,
-    height: 3,
-    borderRadius: 2,
-    marginLeft: 5,
-    marginRight: 5,
-    backgroundColor: colors.black,
-    alignSelf: 'center'
-  },
-  timeText: {
-    fontFamily: fonts.inter[200],
-    fontSize: 10,
-    lineHeight: 12.19,
-    alignSelf: 'center'
-  },
-  mr8: {
-    marginRight: 8
-  }
-});
+export const replyIcon = () => {
+  return (
+    <View style={styles.containerReply}>
+      <View style={styles.containerReplyIcon}>
+        <Icon name={'reply'} size={20} color={colors.black} />
+      </View>
+    </View>
+  );
+};
 
 const ChatItemTargetText = ({
   username = 'Anonymous User',
   time = '4h',
   isContinuous = false,
   message = '',
-  avatar
+  avatar,
+  chatType,
+  messageType,
+  data
 }: ChatItemMyTextProps) => {
+  const {setReplyPreview, onContextMenuPressed} = useMessageHook();
+
+  const swipeableRef = React.useRef<Swipeable | null>(null);
+  const bubblePosition = useSharedValue(0);
+
+  const isReply = messageType === 'reply';
+  const isReplyPrompt = messageType === 'reply_prompt';
+  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
+
+  const animatedBubbleStyle = useAnimatedStyle(() => ({
+    transform: [{translateX: bubblePosition.value}]
+  }));
+
+  const contextMenuActions = [
+    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
+    {title: 'Copy Message', systemIcon: 'square.on.square'}
+  ];
+
   const renderAvatar = React.useCallback(() => {
-    if (isContinuous) return <View style={styles.avatar} />;
+    if (!isShowUserInfo) return <View style={styles.avatar} />;
     return <View style={styles.mr8}>{avatar}</View>;
   }, []);
 
+  const onSwipeToReply = (direction) => {
+    if (direction === 'right') return;
+    if (swipeableRef.current) swipeableRef.current?.close();
+    setReplyPreview({
+      username: 'Anonymous User',
+      time: '4h',
+      message,
+      messageId: 'sdfs',
+      chatType: '',
+      messageType: 'regular'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onSwipeToReply` function has hardcoded values for username, time, messageId, and chatType when calling `setReplyPreview`. These should be replaced with actual values from the message data.
```diff
- setReplyPreview({
-   username: 'Anonymous User',
-   time: '4h',
-   message,
-   messageId: 'sdfs',
-   chatType: '',
-   messageType: 'regular'
- });
+ setReplyPreview({
+   username,
+   time,
+   message,
+   messageId: data?.id,
+   chatType,
+   messageType
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->